server/Transaction.java

```diff
@@ -0,0 +1,236 @@
+                newBankClientHandler.sendOutput("Invalid input, please try again:");
+            }
+        }
+        return null;
+    }
+
+    private Customer getToCustomer(CustomerID customer, NewBankClientHandler newBankClientHandler){
+        String input;
+        Customer payee;
+        valid = false;
+        while (valid == false) {
+            //asks for customer name
+            input = newBankClientHandler.getInput();
+            //check if quit
+            if (checkQuitInput(input)) {return null;}
+
+            //checks if customer exists, if does returns Customer Object
+            //else loops
+            if (NewBank.getBank().getCustomers().containsKey(input)){
+                valid = true;
+                payee = NewBank.getBank().getCustomers().get(input);
+                return payee;
+            }else{
+                newBankClientHandler.sendOutput("No customer found, please try again:");
+            }
+        }
+        return null;
+    }
+
+    private Account getToCustomerAccount(NewBankClientHandler newBankClientHandler){
+        String input;
+        Account account;
+        valid = false;
+
+        while (valid == false) {
+            //asks for payee account input
+            input = newBankClientHandler.getInput();
+            //check if quit
+            if (checkQuitInput(input)) {return null;}
+
+            //if account exists, returns the Account Object
+            //else error message and loops
+            if (toCustomer.checkExistingAccount(input)){
+                valid = true;
+                return toCustomer.getExistingAccount(input);
+            }else{
+                newBankClientHandler.sendOutput("Invalid input, please try again:");
+            }
+        }
+        return null;
+    }
+
+    //
+    public void transferFunds(double amount, Account fromAccount, Account toAccount){
+        //subtracts amount from the 'from' account
+        double fromBalance = fromAccount.getOpeningBalance();
+        fromAccount.setOpeningBalance(fromBalance - amount);
+        //add amount to the 'to' account
+        double toBalance = toAccount.getOpeningBalance();
+        toAccount.setOpeningBalance(toBalance + amount);
+    }
+
+    private boolean checkDouble(String s){
+
+        //tests if input string is of double format
+        try {
+            Double.parseDouble(s);
+        } catch (NumberFormatException e) {
+            return false;
+        }
+        return true;
+    }
+
+    //if sufficient funds, return true else false
+    private boolean checkSufficientFunds(double amount, Account fromAccount) {
+        return !((fromAccount.getOpeningBalance() - amount) < 0);
+    }
+
+    //if argument is negative return true
+    private boolean checkNegative(double d){
+        return d < 0;
+    }
+
+    private boolean checkQuitInput(String s){
+        return s.equalsIgnoreCase("q");
+    }
+
+    private boolean getYN(NewBankClientHandler newBankClientHandler){
```

Comment:
A comment would actually be useful here, as it's not immediately obvious what the author wanted to do